src/ansiblelint/rules/VariableNamingRule.py

```diff
@@ -0,0 +1,117 @@
+import re
+from typing import TYPE_CHECKING, Any, Dict, Generator, List, Tuple, Union
+
+from ansiblelint.file_utils import Lintable
+from ansiblelint.rules import AnsibleLintRule
+from ansiblelint.utils import parse_yaml_from_file
+
+if TYPE_CHECKING:
+    from ansiblelint.constants import odict
+    from ansiblelint.errors import MatchError
+
+
+# properties/parameters are prefixed and postfixed with `__`
+def is_property(k: str) -> bool:
+    """Check if key is a property."""
+    return k.startswith('__') and k.endswith('__')
+
+
+def is_invalid_variable_name(text: str) -> bool:
+    """Check if variable name is using right pattern."""
+    patterns = '^[a-z0-9_]*$'
```

Comment:
Should this be `+` instead of `*`?  Unless you have a check somewhere else for an empty string?

Reply:
FYI, what ansible uses internally https://github.com/ansible/ansible/blob/devel/lib/ansible/utils/vars.py#L279